lib/Test/Class/Moose.pm

```diff
@@ -159,6 +159,9 @@ sub import {
     if ( my $parent = ( delete $args{parent} || delete $args{extends} ) ) {
         my @parents = 'ARRAY' eq ref $parent ? @$parent : $parent;
         $caller->meta->superclasses(@parents);
+        grep { $_->isa(__PACKAGE__) } @parents
```

Comment:
I'd rather see this written as:

```perl
unless ( ... ) {
    ... # construct a $msg
    Carp::croak($msg);
}
```

I think the check itself can be simplified to `unless ( $caller->isa(__PACKAGE__) )`. After we set superclasses the `->isa` method should do the right thing here.

The error message can be improved to give more information (and for grammar):

* Show the entire inheritance chain using `$caller->meta->linearized_isa`.
* If there is only one ancestor, say `"The $caller class inherits from $ancestor, which does not inherit from " . __PACKAGE__`
* For two ancestors, `"The $caller class has X and Y as ancestors but none of these classes inherit from " . 
__PACKAGE__`
* For more than two ancestors, `"The $caller class has X, Y, and Z as ancestors but none of these classes inherit from " . __PACKAGE__`